auth.py

```diff
@@ -122,30 +118,30 @@ def check_auth(self, token_hash, allowed_roles, resource, method):
 def admin_users():
     from argparse import ArgumentParser
     a = ArgumentParser()
-    for op in ('add', 'remove', 'reset'):
-        a.add_argument('--' + op, nargs='+', type=str, default=())
+    a.add_argument('action', choices=('add', 'remove', 'reset'))
+    a.add_argument('username')
+    a.add_argument('--password', nargs='?', default=None)
     args = a.parse_args()
 
-    def _add_user(username):
-        cursor.execute('INSERT INTO users VALUES (?, ?)', (username, hash_pw(username)))
+    def _add_user(username, password=None):
+        cursor.execute('INSERT INTO users VALUES (?, ?)', (username, hash_pw(username or password)))
```

Comment:
Shouldn't the hashing be done on `password or username` ?
here you will always has the username because it can't be `None`